book/src/appendices/b_migration_notes/transform_api_changes.md

```diff
@@ -0,0 +1,175 @@
+
+## Summary
+
+
+* `Float` type has been added.
+* `GlobalTransform` has been removed.
+* `set` methods have been renamed using `set_translation`
+* `local` methods have been renamed using `append`
+* `pitch`, `yaw`, and `roll` methods now use x, y, and z axis respectively.
+* `global` methods have been renamed using `prepend`
+* `set_position` has been renamed `set_translation`
+* 2D helper aliases have been added
+* `set_rotation` methods have been added
+* `euler_angles` has been added
+
+## `Float` Type
+
+A wrapper type around `f32` and `f64`. It is used to hide the actual type being used internally. Mostly used with the Transform type. The default type is `f32` and you can switch to the `f64` type by enabling the "float64" feature gate.
+
+## Renamed Transform Methods (Breaking Changes)
+
+### `set(x|y|z)` to `set_translation(x|y|z)`
+
+```patch
+-transform.set_x(0.2);
++transform.set_translation_x(0.2);
+```
+
+```patch
+-transform.set_y(7.1);
++transform.set_translation_y(7.1);
+```
+
+```patch
+-transform.set_z(2.4);
++transform.set_translation_z(2.4);
+```
+
+```patch
+-transform.set_xyz(0.2, 1.0, 0.8);
++transform.set_translation_xyz(0.2, 1.0, 0.8);
+```
+
+### `_local` to `append_`
+
+```patch
+-transform.move_local(Vector3::new(5.0, 2.0, 3.0));
++transform.append_translation(Vector3::new(5.0, 2.0, 3.0));
+```
+
+```patch
+-transform.move_along_local(Vector3::new(3.0, 2.0, 1.0), 4.0);
++transform.append_translation_along(Vector3::new(3.0, 2.0, 1.0), 8.0);
+```
+
+```patch
+-transform.rotate_local(Vector3::new(2.0, 2.1, 3.0), 8.0);
++transform.append_rotation(Vector3::new(2.0, 2.1, 3.0), 8.0);
+```
+
+### `(pitch|yaw|roll)_local` to `append_rotation_(x|y|z)_axis`
+
+```patch
+-transform.pitch_local(2.3);
++transform.append_rotation_x_axis(2.3);
+```
+
+```patch
+-transform.yaw_local(1.0);
++transform.append_rotation_y_axis(1.0);
+```
+
+```patch
+-transform.roll_local(2.3);
++transform.append_rotation_z_axis(4.6);
+```
+
+### `_global` to `prepend_`
+
+```patch
+-transform.move_global(Vector3::new(5.0, 2.0, 3.0));
++transform.prepend_translation(Vector3::new(5.0, 2.0, 3.0));
+```
+
+```patch
+-transform.move_along_global(Vector3::new(5.0, 2.0, 3.0));
++transform.prepend_translation_along(Vector3::new(5.0, 2.0, 3.0));
+```
+
+```patch
+-transform.rotate_global(Vector3::new(0.2, 0.4, 0.6), 0.4);
++transform.prepend_rotation(Vector3::new(0.2, 0.4, 0.6), 0.4);
+```
+
+### `(pitch|yaw|roll)_global` to `append_rotation_(x|y|z)_axis`
+
+```patch
+-transform.pitch_global(0.4);
++transform.prepend_rotation_x_axis(0.4);
+```
+
+```patch
+-transform.yaw_global(0.4);
++transform.prepend_rotation_y_axis(0.4);
+```
+
+```patch
+-transform.roll_global(0.4);
++transform.prepend_rotation_z_axis(0.4);
+```
+
+### `translate_[xyz]` to `prepend_translation_[xyz]`
```

Comment:
hm, could you use either `(x|y|z)` here, or square brackets for the other points